include/fiction/algorithms/physical_design/on_the_fly_circuit_design_on_defective_surface.hpp

```diff
@@ -0,0 +1,216 @@
+//
+// Created by Jan Drewniok on 26.06.24.
+//
+
+#ifndef FICTION_ON_THE_FLY_CIRCUIT_DESIGN_ON_DEFECTIVE_SURFACE_HPP
+#define FICTION_ON_THE_FLY_CIRCUIT_DESIGN_ON_DEFECTIVE_SURFACE_HPP
+
+#include "fiction/algorithms/physical_design/apply_gate_library.hpp"
+#include "fiction/algorithms/physical_design/exact.hpp"
+#include "fiction/networks/technology_network.hpp"
+#include "fiction/technology/parameterized_gate_library.hpp"
+#include "fiction/technology/sidb_defect_surface.hpp"
+#include "fiction/technology/sidb_skeleton_bestagon_library.hpp"
+#include "fiction/technology/sidb_surface_analysis.hpp"
+#include "fiction/traits.hpp"
+#include "fiction/types.hpp"
+
+#include <mockturtle/algorithms/equivalence_checking.hpp>
+#include <mockturtle/algorithms/miter.hpp>
+#include <mockturtle/utils/stopwatch.hpp>
+
+#include <exception>
+#include <iostream>
+#include <optional>
+#include <utility>
+
+namespace fiction
+{
+
+/**
+ * This struct stores the parameters to design SiDB circuit on a defective surface.
+ *
+ * @tparam CellLyt Cell-level layout type.
+ */
+template <typename CellLyt>
+struct on_the_fly_circuit_design_params
+{
+    /**
+     * Parameters for the parameterized gate library.
+     */
+    parameterized_gate_library_params<CellLyt> parameterized_gate_library_parameter = {};
+    /**
+     * Parameters for the *exact* placement&routing algorithm.
+     */
+    exact_physical_design_params exact_design_parameter = {};
+};
+
+/**
+ * Statistics for the on-the-fly defect-aware circuit design.
+ */
+struct on_the_fly_circuit_design_stats
+{
+    /**
+     * The total runtime of the operational domain computation.
+     */
+    mockturtle::stopwatch<>::duration time_total{0};
+    /**
+     * The `stats` of the *exact* algorithm.
+     */
+    exact_physical_design_stats exact_stats{};
+};
+
+namespace detail
+{
+
+template <typename Ntk, typename CellLyt, typename GateLyt>
+class on_the_fly_circuit_design_impl
+{
+  public:
+    on_the_fly_circuit_design_impl(const Ntk& network, const on_the_fly_circuit_design_params<CellLyt>& design_params,
+                                   const GateLyt& tiling, on_the_fly_circuit_design_stats& st) :
+            lattice_tiling{tiling},
+            mapped_network{network},
```

Comment:
Why mapped?

Reply:
I apologize for these kinds of mistakes. I'm not that familiar with these details yet. But now I understand that it doesn't necessarily have to be mapped. Thank you!